include/alpaka/atomic/AtomicAtomicRef.hpp

```diff
@@ -12,13 +12,27 @@
 #include <alpaka/atomic/Traits.hpp>
 #include <alpaka/core/BoostPredef.hpp>
 
-#include <boost/atomic.hpp>
+#if defined(ALPAKA_HAS_STD_ATOMIC_REF)
```

Comment:
why not simply
```suggestion
 #if defined(__cpp_lib_atomic_ref)
```
?

Reply:
I want to avoid unnecessary includes. `__cpp_lib_atomic_ref` is only defined **after** including `<atomic>`.

Reply:
you could `#include <version>`

Reply:
That would work. However, it would mean that this:

```c++
#if defined(ALPAKA_HAS_STD_ATOMIC_REF)
#    include <atomic>
```

turns into this:

```c++
#if __cplusplus >= 202002L
#    include <version>
#endif

#if defined(__cpp_lib_atomic_ref)
#    include <atomic>
```

which I wanted to avoid. And since I needed to change the way CMake picks up the dependency anyway (we don't want to link to Boost.Atomic if that is unnecessary) I introduced the macro `ALPAKA_HAS_STD_ATOMIC_REF`.

Once we have a separate requirement header (see #1636) we can move the checks into there. And that would look pretty similar to the second approach.